lucene/core/src/java/org/apache/lucene/util/automaton/Automata.java

```diff
@@ -254,7 +254,7 @@ public static Automaton makeBinaryInterval(BytesRef min, boolean minInclusive, B
       cmp = min.compareTo(max);
     } else {
       cmp = -1;
-      if (min.length == 0 && minInclusive) {
+      if (min.length == 0) {
```

Comment:
this looks wrong as we should still make sure that the empty string is rejected if min=="" and minInclusive==false?